### PR TITLE
Test transcription shielding while retaining clean audio

### PR DESCRIPTION
**Pull request by ComputelessComputer**
Add an isolated local experiment with model and codec comparisons, listening reports, and measurement checks. ANLG-393.

---
## Summary by cubic
Implements ANLG-393 with an isolated, offline transcription-shield experiment. It fits a perturbation to one complete utterance while retaining the clean audio, then compares transcription and codec effects through reproducible artifacts.

**Details**

- Adds WER, audio validation, SNR and length checks, deterministic noise control, safe WAV writing, and escaped HTML report generation.
- Makes noise-control comparisons reproducible with a fixed seed and by rejecting shuffled perturbations that would clip instead of changing their distribution.
- Compares `tiny.en`, `base.en`, or `small.en` models before and after a 24 kbps Opus round trip.
- Runs with `uv`, requires `ffmpeg`, downloads Whisper models locally, and refuses existing output directories.
- Adds tests for metric edge cases, clipping, WAV round trips, deterministic perturbations, and report escaping.
- Limits the experiment to one utterance and Whisper-based listening and measurement; it does not evaluate real-time calls or human intelligibility.

<sup>Written for commit 8190a5b2514dec65900cf300a79994c3a2406491. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/fastrepl/anarlog/pull/7682?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: scripts/test_transcription_shield.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import tempfile
+import unittest
+import wave
+from pathlib import Path
+
+import numpy as np
```
[the remark this answers is not in the document]
Comment (ComputelessComputer) [reply scripts/test_transcription_shield.py:12]: Fixed in 47b538df20: the test now declares its Python and dependency versions in PEP 723 metadata and documents uv run --script scripts/test_transcription_shield.py. That command passes all seven tests in its isolated declared environment.

File: scripts/transcription_shield.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        else None,
+        "peak_delta": float(np.max(np.abs(delta))),
+        "length_matches": True,
+    }
+
+
+def noise_control(clean, shield, seed):
+    # Shuffling preserves the perturbation's energy and amplitude distribution.
+    delta = np.random.default_rng(seed).permutation(shield - clean)
+    return np.clip(clean + delta, -1, 1)
```
[the remark this answers is not in the document]
Comment (ComputelessComputer) [reply scripts/transcription_shield.py]: Fixed in 47b538df20: noise_control rejects shuffled waveforms that exceed full scale instead of clipping them. Tests verify deterministic distribution/energy preservation on accepted inputs and rejection when a permutation would clip; all seven tests pass.
